src/reducers/searchReducer.js

```diff
@@ -0,0 +1,83 @@
+const initialState = {
+  searchQuery: '',
+  page: 1,
+  displayFields: {
+    categories: 'show',
+    authors: 'show',
+    tags: 'show',
+  },
+  updateFields: {
+    categories: [],
+    authors: [],
+    tags: [],
+  },
+  searchResults: [],
+};
+
+const filterReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case 'UPDATE_FILTER': {
+      const { field, value } = action.payload;
+      return {
+        ...state,
+        updateFields: {
+          ...state.updateFields,
+          [`${field}`]: state.updateFields[`${field}`].concat([value]),
+        },
+      };
+    }
+
+    case 'DISPLAY_FILTER': {
+      const value = state.displayFields[`${action.payload}`] === 'hide' ? 'show' : 'hide';
+      return {
+        ...state,
+        displayFields: {
+          ...state.displayFields,
+          [`${action.payload}`]: value,
+        },
+      };
+    }
+
+    case 'REMOVE_FILTER': {
+      const { field, value } = action.payload;
+      return {
+        ...state,
+        updateFields: {
+          ...state.updateFields,
+          [`${field}`]: state.updateFields[`${field}`].filter((item) => item !== value),
+        },
+      };
+    }
+
+    case 'UPDATE_ARTICLES': {
+      const { payload } = action;
+      return {
+        ...state,
+        searchResults: [
+          ...payload,
+        ],
+      };
+    }
+
+    case 'UPDATE_SEARCHQUERY': {
+      const { payload } = action;
+      return {
+        ...state,
+        searchQuery: payload,
+      };
+    }
+
+    case 'UPDATE_PAGENUMBER': {
```

Comment:
This actions should not be defined here, cause it has already been defined in the action types. 
which should be referenced here instead like so `actionType. UPDATE_PAGENUMBER` they are constants.

Reply:
Please make the necessary changes for all the defined action in this file and other files.

Reply:
That's true. I have made the necessary changes. Thanks